### PR TITLE
Fixed token scope for Azure Government ADX access

### PR DESCRIPTION
**Pull request by floushee**
We run this version of the plugin for customers in the public and government cloud and thus tested it in production. I can't speak for the China Cloud as I don't have access to such environment.

### CODE REVIEW

**Comment by sunker** [comment]
Thanks for contributing @floushee! I'm not sure this is the right approach. @vladikbr WDYT?

File: pkg/azuredx/datasource.go [hunk 1]
```diff
@@ -36,7 +35,11 @@ func NewDatasource(settings backend.DataSourceInstanceSettings) (instancemgmt.In
 	}
 	adx.settings = datasourceSettings
 
-	tokenProvider := tokenprovider.NewAccessTokenProvider(tokenCache, datasourceSettings.ClientID, datasourceSettings.TenantID, datasourceSettings.AzureCloud, datasourceSettings.Secret, []string{"https://kusto.kusto.windows.net/.default"})
+	adxScope := "https://kusto.kusto.windows.net/.default"
+	if datasourceSettings.AzureCloud == "govazuremonitor" {
```
Comment (sunker) [inline pkg/azuredx/datasource.go:39]: Please use the constant in clouds.go

**Review by sunker — CHANGES_REQUESTED** [review]

**Comment by floushee** [comment]
I have changed the fork to use the constants and also added the china cloud settings based on the information from #244 .

File: pkg/azuredx/datasource.go (hunk 1 above)
Comment (floushee) [reply pkg/azuredx/datasource.go]: I changed the implementation accordingly.

**Comment by sunker** [comment]
Thanks @floushee! I now have contact person at Microsoft that can help us test this in national clouds. Would you mind resolving the merge conflict so that I can ask him to help out?

**Comment by floushee** [comment]
@sunker it should be resolved :) and great that we get help to review this!

**Comment by floushee** [comment]
Any news on this?

**Comment by sunker** [comment]
Yes! Microsoft it helping out testing this. The last thing I heard was that it works as expected in the USGov cloud. I'm just waiting for verification that it also works in China cloud.

**Comment by floushee** [comment]
@sunker  any news on the china cloud test?

**Comment by sunker** [comment]
No. I'll ask them for an update today.

**Comment by sunker** [comment]
> No. I'll ask them for an update today.

Asked MS for a status update two weeks ago. Still waiting for a response.

**Comment by andresmgot** [comment]
This was finally merged in https://github.com/grafana/azure-data-explorer-datasource/pull/258 so closing this PR

**Pull request closed without merging**